### ISSUE TITLE
[FR] Update ndk-gdb to work with gradle/cmake

### ISSUE THREAD

**Issue by safield** [android-ndk/ndk#1024]
In older versions of the Android NDK I could debug my code from the command line using ndk-gdb script. Now that the NDK is migrated to gradle/cmake, the ndk-gdb script is no longer working. This is because it only works with the old build system, and is dependent on the android.mk file.

It would be really nice if there was a new ndk-gdb that worked with the cmake build system and used lldb.

I am a text editor/command line developer and prefer not to open Android Studio for any reason :P

Thanks! :)

**Comment by DanAlbert** [comment]
Yeah, agreed that this is a thing we should do. Specifically, we should keep the existing the existing behavior but add support for:

1. working with gradle projects
2. working with cmake-based gradle projects
3. (optionally, eventually by default?) using lldb behind the scenes

I don't know when we'll have a chance to do this (maybe it's something @hhb wants to look at soon, since he's already poking around with debugger things?), but it's definitely on the to do list.

**Comment by enh-google** [comment]
> (optionally, eventually by default?) using lldb behind the scenes

fwiw, ndk-gdb does now default to lldb. slightly older versions had a --lldb option to request lldb.
